compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/DecomposeAttention.cpp

```diff
@@ -554,99 +338,40 @@ void decomposeTiledAttention(IREE::LinalgExt::AttentionOp tiledAttnOp,
-/// 2. Initialize statistics: running_sum, running_max
-/// 3. for i = 0 to S with step T
-///    a. Load a tile from the K matrix of size T x d -> k
-///    b. Load a tile from the V matrix of size T x d -> v
-///    c. Compute matmul_transpose_b(q, k) -> qkT
-///    d. Compute max(max(qkT) along rows, old_max) -> new_max
-///    e. Compute curent estimate of softmax: exp(qKT - current_max) -> s
-///    f. Compute product of fixup and old_sum -> fsum
-///    g. Compute sum(sum(qkT) along rows, fsum) -> new_sum
-///    h. Compute 1.0 / new_sum -> inv_new_sum
-///    i. Compute softmax = softmax * inv_new_sum
-///    j. Truncate softmax to fp16
-///    k. Compute fsum  * inv_new_sum * accumulator -> new_accumulator
-///    j. Compute matmul(s, v) and add new_accumulator
-///
-///
-LogicalResult reifyAttentionTransform(mlir::FunctionOpInterface funcOp,
-                                      bool onlyTile,
-                                      std::optional<uint64_t> tileSize) {
-  IRRewriter rewriter(funcOp.getContext());
-  funcOp.walk([&](IREE::LinalgExt::AttentionOp attnOp) {
-    SmallVector<Operation *> ops;
-    tileAndDecomposeAttention(attnOp, ops, rewriter, onlyTile, tileSize);
-    return WalkResult::advance();
-  });
-  return success();
-}
-
-} // namespace
-
-namespace {
-struct TileAndDecomposeAttentionPass
-    : public TileAndDecomposeAttentionBase<TileAndDecomposeAttentionPass> {
+struct DecomposeAttentionPass
+    : public DecomposeAttentionBase<DecomposeAttentionPass> {
   void getDependentDialects(DialectRegistry &registry) const override {
     registry.insert<
         affine::AffineDialect, IREE::LinalgExt::IREELinalgExtDialect,
         linalg::LinalgDialect, scf::SCFDialect, tensor::TensorDialect>();
   }
-  TileAndDecomposeAttentionPass() = default;
-  TileAndDecomposeAttentionPass(bool onlyTile, uint64_t tileSize) {
-    this->onlyTile = onlyTile;
+  DecomposeAttentionPass() = default;
+  DecomposeAttentionPass(bool onlyTile, uint64_t tileSize) {
     this->tileSize = tileSize;
   }
-  TileAndDecomposeAttentionPass(const TileAndDecomposeAttentionPass &pass) {
-    onlyTile = pass.onlyTile;
+  DecomposeAttentionPass(const DecomposeAttentionPass &pass) {
     tileSize = pass.tileSize;
   }
   void runOnOperation() override;
 };
 } // namespace
 
-void TileAndDecomposeAttentionPass::runOnOperation() {
+void DecomposeAttentionPass::runOnOperation() {
   MLIRContext *context = &getContext();
   IRRewriter rewriter(context);
   std::optional<uint64_t> optionalTileSize{std::nullopt};
   if (tileSize.hasValue()) {
     optionalTileSize = tileSize.getValue();
   }
-  if (failed(reifyAttentionTransform(getOperation(), onlyTile,
-                                     optionalTileSize))) {
-    return signalPassFailure();
-  }
+  getOperation().walk([&](AttentionOp attnOp) {
+    SmallVector<Operation *> ops;
+    decomposeTiledAttention(attnOp, ops, rewriter, optionalTileSize);
```

Comment:
Do we still need tileSize to decompose the attention op?

Reply:
For some reason I also cannot understand, yes. The logic for decomposition will anyway be changed in the next patch and this can be removed.